### PR TITLE
Update go client version to 0.8.1 for Go Function

### PR DESCRIPTION
**Pull request by wolfstudy**
Signed-off-by: xiaolongran <xiaolongran@tencent.com>

### Motivation

Update go client version to 0.8.1 for Go Function

### Modifications

- Update `go.mod` file
- Upgrade zlib version to 1.2.12

### CODE REVIEW

File: pulsar-function-go/go.mod [hunk 1]
```diff
@@ -3,15 +3,16 @@ module github.com/apache/pulsar/pulsar-function-go
 go 1.13
 
 require (
-	github.com/apache/pulsar-client-go v0.7.0
-	github.com/golang/protobuf v1.4.3
-	github.com/prometheus/client_golang v1.7.1
+	github.com/apache/pulsar-client-go v0.8.0
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
+	github.com/golang/protobuf v1.5.2
+	github.com/prometheus/client_golang v1.11.0
```
Comment (nicoloboschi) [inline pulsar-function-go/go.mod:9]: 1.11.0 is flagged with a critical CVE, could you upgrade it to 1.11.1 ?

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: pulsar-function-go/go.mod (hunk 1 above)
Comment (wolfstudy) [reply pulsar-function-go/go.mod]: Good ideas, will upgrade go sdk version to 081

**Review by nicoloboschi — APPROVED** [review]

**Comment by wolfstudy** [comment]
image
The action CI error

**Review by Shoothzj — APPROVED** [review]

File: pulsar-function-go/go.mod [hunk 2]
```diff
@@ -3,15 +3,16 @@ module github.com/apache/pulsar/pulsar-function-go
 go 1.13
 
 require (
-	github.com/apache/pulsar-client-go v0.7.0
-	github.com/golang/protobuf v1.4.3
-	github.com/prometheus/client_golang v1.7.1
+	github.com/apache/pulsar-client-go v0.8.1
+	github.com/dgrijalva/jwt-go v3.2.0+incompatible // indirect
```
Comment (lhotari) [inline pulsar-function-go/go.mod:7]: There's an existing advisory about this library: https://github.com/advisories/GHSA-w73w-5m7g-f7qc . Can we migrate to https://github.com/golang-jwt/jwt ?

**Review by lhotari — CHANGES_REQUESTED** [review]

File: pulsar-function-go/go.mod (hunk 2 above)
Comment (lhotari) [reply pulsar-function-go/go.mod:7]: https://github.com/apache/pulsar-client-go/blob/3e1b39dd6f88aa0a12b949b1d1e064bde6355e22/oauth2/go.mod#L7 seems to be used in pulsar-client-go/oauth2

Comment (wolfstudy) [reply pulsar-function-go/go.mod:7]: The Go SDK does not directly reference this lib. This is indirectly referenced by other packages, and we cannot modify it.

**Review by aloyszhang — APPROVED** [review]

File: pulsar-function-go/go.mod (hunk 2 above)
Comment (wolfstudy) [reply pulsar-function-go/go.mod:7]: Hello @lhotari i send a pull request to Go SDK, refer to: https://github.com/apache/pulsar-client-go/pull/759, please check.

Comment (wolfstudy) [reply pulsar-function-go/go.mod]: @lhotari PTAL again, thanks

**Review by lhotari — APPROVED** [review]
LGTM

**Pull request merged**